File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/mr/nu/obstetrics/breathmachnur/d/BreathmachnurAggDO.java
package iih.ci.mr.nu.obstetrics.breathmachnur.d;

import xap.sys.appfw.orm.model.agg.BaseAggDO;
import xap.mw.core.data.BaseDO;
import xap.mw.coreitf.i.IAggDesc;
import xap.mw.coreitf.i.IBaseDO;
import iih.ci.mr.nu.obstetrics.breathmachnur.d.desc.BreathmachnurAggDODesc;



/**
 * 呼吸机治疗观察记录单
 */
public class BreathmachnurAggDO extends BaseAggDO {
	private static final long serialVersionUID = 1L;

	@Override
	public BreathMachInfoDO getParentDO() {
		return ((BreathMachInfoDO) super.getParentDO());
	}

	public void setParentDO(BreathMachInfoDO headDO) {
		setParent(headDO);
	}

	public BreathMachRecDO[] getBreathMachRecDO() {
		IBaseDO[] dos = getChildren(BreathMachRecDO.class);
		if(dos==null) return null;
		BreathMachRecDO[] result=new BreathMachRecDO[dos.length];
		for(int i=0;i<dos.length;i++){
			result[i]=(BreathMachRecDO)dos[i];
		}
		return result;
	}
	
	public void setBreathMachRecDO(BreathMachRecDO[] dos) {
		setChildren(BreathMachRecDO.class, dos);
	}

	@Override
	public IAggDesc getAggDesc() {
		BreathmachnurAggDODesc desc = new BreathmachnurAggDODesc();

		return desc;
	}
	
	 public  BaseDO createParentDO() {
         return new BreathMachInfoDO();
     }

     public  BaseDO createChildDO(String clzName) {
	     if (clzName.equals("iih.ci.mr.nu.obstetrics.breathmachnur.d.BreathMachRecDO")) {
             return new BreathMachRecDO();
         }
         return null;
     }
}

File: client/iih_stub/iih.ci_stub/iih.ci.mr/nu/obstetrics/breathmachnur/d/BreathmachnurAggDO.cs
using System;
using xap.mw.core.data;
using xap.mw.core.meta;

namespace iih.ci.mr.nu.obstetrics.breathmachnur.d
{
	/// <summary>
	/// 呼吸机治疗观察记录单DO
	/// </summary>
	public class BreathmachnurAggDO : BaseAggDO
	{	
	    private static string BREATHMACHINFODO_BREATHMACHRECDO="iih.ci.mr.nu.obstetrics.breathmachnur.d.BreathMachRecDO";

        public BreathmachnurAggDO() {
            this.setParentDO(new BreathMachInfoDO());
        }

  	    public BreathMachInfoDO getParentDO() {
		    return ((BreathMachInfoDO) base.ParentDO);
	    }      
	    
	    public void setParentDO(BreathMachInfoDO headDO) {
		    base.ParentDO=headDO;
	    }
	    
	    public BreathMachRecDO[] getBreathMachRecDO() {		    
		    
		    BaseDO[] dos=getChildren(Type.GetType(BREATHMACHINFODO_BREATHMACHRECDO));
		    if (dos == null || dos.Length==0){return null;}
            BreathMachRecDO[] r = new BreathMachRecDO[dos.Length];
            for (int i = 0; i < dos.Length;i++ ){
                r[i] = dos[i] as BreathMachRecDO;
            }
	        return r;
		    
	    }
	    
	    public void setBreathMachRecDO(BreathMachRecDO[] dos) {
            if (dos == null || dos.Length == 0) return;
            setChildren(dos[0].GetType(), dos);
	    }


	    public override BaseDO createParentDO()
	    {
	        return new BreathMachInfoDO();
	    }

        public override BaseDO createChildDO(string clzName)
	    {
            if (BREATHMACHINFODO_BREATHMACHRECDO == clzName)
	        {
                return new BreathMachRecDO();
	        }
	        return null;
	    }
	    
	    /// <summary>
        /// 获取子类全路径名数组
        /// </summary>
        /// <returns></returns>
        public override string[] getChildClassNames(){
            return  new string[]{"iih.ci.mr.nu.obstetrics.breathmachnur.d.BreathMachRecDO"};
        }
	    public new static AggMeta GetMeta()
	    {
	        AggMeta meta = new AggMeta();
	        meta.OneToOneDOs = new string[]{};
	        return meta;
	    }
	    
	}
}
